boards: nrf9161dk_nrf9161_ns: Add TF-M support to nrf9161 DK

joerchan opened the pull request.

joerchan committed:
boards: nrf9161dk_nrf9161_ns: Add TF-M support to nrf9161 DK

Add TF-M support for nrf9161 DK and enable it by default for the
non-secure board variant.
Disable UART1 since TF-M use this for output and it is configured
as a secure peripheral.
Enabling this will trigger a BusFault in TF-M.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
manifest: Update trusted-firmware-m for nrf9161 DK support

Update trusted-firmware-m to add nrf9161 DK support

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

SebastianBoe reviewed (changes requested):
IMHO we should add this board to a subset of the TF-M sample.yaml's for regression testing purposes.

Also, I don't see any doc changes.

I see from the 9161 DK docs that there is a note about TF-M being unsupported.

I suppose we should change these docs now?

https://docs.zephyrproject.org/latest/boards/arm/nrf9161dk_nrf9161/doc/index.html#building-secure-non-secure-zephyr-applications-with-arm-reg-trustzone-reg

joerchan committed:
boards: nrf9161dk_nrf9161: Remove TF-M not supported note

Remove note nrf9161dk_nrf9161 board stating that TF-M is not supported
for this board.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
samples: tfm_integration: Add nrf9161dk_nrf9161_ns to TF-M test samples

Add nrf9161dk_nrf9161_ns board to the TF-M integration samples for
testing.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
